main.tf

```diff
@@ -650,7 +650,7 @@ resource "aws_network_acl" "public" {
   count = local.create_vpc && var.public_dedicated_network_acl && length(var.public_subnets) > 0 ? 1 : 0
 
   vpc_id     = local.vpc_id
-  subnet_ids = aws_subnet.public[*].id
+  subnet_ids = [for sid in aws_subnet.public[*].id : sid if !contains(var.public_network_acl_excluded_subnets, sid)]
```

Comment:
you cannot pass subnet IDs that you do not yet know their value - this is incorrect

Reply:
In general, yes, but here, it's for a special use case where one has executed the VPC module first to create the resources, and afterwards simulate an AZ outage for a specific subnet.

The best way to target without any ambiguity a subnet is through the subnet ID.

We could offer as an alternative filter based on tags also, but the filter by subnet ID should still be an option, I think.

Reply:
>  but here, it's for a special use case

And this is why I don't believe this type of change belongs in the module

Reply:
How could we test an AZ outage then? Isn't it a use case that the VPC module would/should support? Or is there any other way to test AZ outage?

I'm all ears, if there's an alternative.